beacon_node/rest_api/src/validator.rs

```diff
@@ -147,3 +134,189 @@ pub fn get_validator_duties<T: BeaconChainTypes + 'static>(req: Request<Body>) -
     );
     Ok(success_response(body))
 }
+
+/// HTTP Handler to produce a new BeaconBlock from the current state, ready to be signed by a validator.
+pub fn get_new_beacon_block<T: BeaconChainTypes + 'static>(req: Request<Body>) -> ApiResult {
+    let beacon_chain = get_beacon_chain_from_request::<T>(&req)?;
+
+    let query = UrlQuery::from_request(&req)?;
+    let slot = match query.first_of(&["slot"]) {
```

Comment:
FYI this could be something like

```rust
let slot = query.first_of(&["slot"])
  .map(|_key, value| value)?
  .parse::<u64>()
  .map_err(|e| ApiError::InvalidQueryParams(format!("Invalid slot parameter, must be a u64. {:?}", e)))?
```

Note: I haven't tested this, it probably doesn't compile.